app.py

```diff
@@ -296,84 +338,54 @@ def calculate_biodiversity_score(start_year, end_year, project_name):
             """
             USE climatebase;
             CREATE TABLE IF NOT EXISTS bioindicator (year BIGINT, project_name VARCHAR(255), value DOUBLE, area DOUBLE, score DOUBLE, CONSTRAINT unique_year_project_name UNIQUE (year, project_name));
+        """)
+        # UPSERT project record
+        con.sql(
+            """
+            INSERT INTO bioindicator FROM _temptable
+            ON CONFLICT (year, project_name) DO UPDATE SET value = excluded.value;
         """
         )
-
-    return con.sql(
-        f"SELECT * FROM bioindicator WHERE (year > {start_year} AND year <= {end_year} AND project_name = '{project_name}')"
-    ).df()
-
-
-def view_all():
-    logging.info("view_all")
-    return con.sql(f"SELECT * FROM bioindicator").df()
-
-
-def push_to_md():
-    # UPSERT project record
-    con.sql(
-        """
-        INSERT INTO bioindicator FROM _temptable
-        ON CONFLICT (year, project_name) DO UPDATE SET value = excluded.value;
-    """
-    )
-    logging.info("upsert records into motherduck")
-
+        logging.info("upsert records into motherduck")
+    scores = \
+        con.execute("SELECT * FROM bioindicator WHERE (year >= ? AND year <= ? AND project_name = ?)",
+                    [start_year, end_year, project_name]).df()
+    return scores
 
 def motherduck_list_projects(author_id):
-    return con.sql(
-        f"""
-        SELECT DISTINCT name FROM project WHERE authorId = '{author_id}'
-    """
-    ).df()
+    return \
+        con.execute("SELECT DISTINCT name FROM project WHERE authorId = ? AND geometry != 'null'", [author_id]).df()
```

Comment:
At the moment the DB has null values set to the string 'null' instead of the null character. We should try to avoid this if possible.

Reply:
Is this the notion? I am having trouble finding it in my email or on Slack.